lib/msf/core/modules/loader/base.rb

```diff
@@ -479,10 +479,11 @@ def module_path?(path)
     module_path
   end
 
+  # Tries to determine if a file might be executable,
   def script_path?(path)
     File.executable?(path) &&
       !File.directory?(path) &&
-      File.read(path, 2) == "#!"
+      ['#!', '//'].include?(File.read(path, 2))
```

Comment:
so a file starting with a comment is now treated as executable? Doesn't this cause a lot false positives?

Reply:
This actually says that, in order for a module to be considered executable as an external module, it must be both executable on the filesystem, and contain either a `#!` or a `//` as the first 2 bytes of the file.

This code is only applied to files in the `modules` directory. Neither Python nor Ruby modules should start with `//` because that is not a comment sequence in the first place.

This function originally just checked for the executable bit, but someone then tried running Metasploit from a FAT32 filesystem mounted in Linux, where all files were executable by default. This is just belt-and-suspenders code.

Reply:
You could possibly inject a `//` at the beginning of a Ruby script, and get by syntactically if you put it on its own line, and `chmod +x` it, but this is going to have a much more limited practical scope than it may seem in isolation.

A possible better long-term solution to this issue of deriving which modules are to be treated as _external_ might be to make the metadata an external .json file that lives along-side the script file, and this contains the metadata that tells Metasploit how to run it. @acammack-r7 has an early spec for something like this if you're interested.

Reply:
Yeah, I would love to get rid of this magic, but `//` is safe until we start adding source files in languages that need to be compiled. At that point we'll need instructions for how to build the source anyway, so hopefully we can roll "hey, I run as a separate process" and "hello, here are my LDFLAGS" into the same mechanism.